gcsstore/gcsstore.go

```diff
@@ -32,17 +33,23 @@ type GCSStore struct {
 	// Specifies the GCS bucket that uploads will be stored in
 	Bucket string
 
+	// ObjectPrefix is prepended to the name of each GCS object that is created.
+	// It can be used to create a pseudo-directory structure in the bucket,
+	// e.g. "path/to/my/uploads".
+	ObjectPrefix string
+
 	// Service specifies an interface used to communicate with the Google
 	// cloud storage backend. Implementation can be seen in gcsservice file.
 	Service GCSAPI
 }
 
 // New constructs a new GCS storage backend using the supplied GCS bucket name
 // and service object.
-func New(bucket string, service GCSAPI) GCSStore {
+func New(bucket string, objPrefix string, service GCSAPI) GCSStore {
```

Comment:
Adding the objPrefix argument is unfortunately as breaking change and we would like to avoid them for now. However, I think the caller should be able to set the ObjectPrefix after calling New, so the argument is now needed actually:

```
store := gcsstore.New("bucket", gcsService)
store.ObjectPrefix = "prefix/"
```